### PR TITLE
[MM-28125] einterface/oauthprovide: return parsing errors

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
It's better to return parsing errors and avoid nil pointer dereferences.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28125

### CODE REVIEW

File: app/oauth.go [hunk 1]
```diff
@@ -574,19 +574,14 @@ func (a *App) LoginByOAuth(service string, userData io.Reader, teamId string) (*
...
-	if authUser.AuthData != nil {
-		authData = *authUser.AuthData
-	}
-
-	if len(authData) == 0 {
+	authUser, err1 := provider.GetUserFromJson(bytes.NewReader(buf.Bytes()))
+	if err1 != nil || *authUser.AuthData == "" {
 		return nil, model.NewAppError("LoginByOAuth", "api.user.login_by_oauth.parse.app_error",
-			map[string]interface{}{"Service": service}, "", http.StatusBadRequest)
+			map[string]interface{}{"Service": service}, err1.Error(), http.StatusBadRequest)
```
Comment (wiggin77) [inline app/oauth.go:581]: Is it possible for err1 to be nil and *authUser.AuthData == ""?

Comment (isacikgoz) [reply app/oauth.go:581]: It's a possibility. It escaped from me thanks @wiggin77  :)
```suggestion
			map[string]interface{}{"Service": service}, "", http.StatusBadRequest)
```

**Review by reflog — APPROVED** [review]
LGTM, except for the comment @wiggin77 left

File: app/oauth.go [hunk 2]
```diff
@@ -574,19 +574,14 @@ func (a *App) LoginByOAuth(service string, userData io.Reader, teamId string) (*
...
-	authData := ""
-	if authUser.AuthData != nil {
-		authData = *authUser.AuthData
-	}
-
-	if len(authData) == 0 {
+	authUser, err1 := provider.GetUserFromJson(bytes.NewReader(buf.Bytes()))
+	if err1 != nil || *authUser.AuthData == "" {
 		return nil, model.NewAppError("LoginByOAuth", "api.user.login_by_oauth.parse.app_error",
 			map[string]interface{}{"Service": service}, "", http.StatusBadRequest)
```
Comment (wiggin77) [inline app/oauth.go:581]: We're losing the error details.  Not sure how useful but easy enough to include them but only if err1 != nil

**Comment by amyblais** [comment]
@isacikgoz Will this go to Doug's review next?

**Review by wiggin77 — APPROVED** [review]
LGTM  :+1:

**Review by lindalumitchell — APPROVED** [review]

**Pull request merged**